src/analyses/modifiedSinceLongjmp.ml

```diff
@@ -0,0 +1,92 @@
+(** Locally track the variables that may have been written since the corresponding jumpbuffer was set *)
+
+open Prelude.Ana
+open Analyses
+
+module Spec =
+struct
+  include Analyses.DefaultSpec
+
+  let name () = "modifiedSinceLongjmp"
+  module D = JmpBufDomain.LocallyModifiedMap
+  module VS = D.VarSet
+  module C = Lattice.Unit
+
+  let add_to_all_defined vs d =
+    D.map (fun vs' -> VS.union vs vs') d
+
+  let is_relevant v =
+    (* Only checks for v.vglob on purpose, acessing espaced locals after longjmp is UB like for any local *)
+    not v.vglob (* *) && not (BaseUtil.is_volatile v) && v.vstorage <> Static
+
+  let relevants_from_ls ls =
+    if Queries.LS.is_top ls then
+      VS.top ()
+    else
+      Queries.LS.fold (fun (v, _) acc -> if is_relevant v then VS.add v acc else acc) ls (VS.empty ())
+
+  let relevants_from_lval_opt ctx lval = match lval with
+    | Some (Var v, _) -> if is_relevant v then VS.singleton v else VS.empty ()
+    | Some (Mem e, _) -> relevants_from_ls (ctx.ask (Queries.MayPointTo e))
+    | None -> VS.empty ()
+
+  (* transfer functions *)
+  let assign ctx (lval:lval) (rval:exp) : D.t =
+    if not !Goblintutil.assign_is_setjmp then
+      add_to_all_defined (relevants_from_lval_opt ctx (Some lval)) ctx.local
+    else
+      ctx.local
+
+  let branch ctx (exp:exp) (tv:bool) : D.t =
+    ctx.local
+
+  let body ctx (f:fundec) : D.t =
+    ctx.local
+
+  let return ctx (exp:exp option) (f:fundec) : D.t =
+    ctx.local
+
+  let enter ctx (lval: lval option) (f:fundec) (args:exp list) : (D.t * D.t) list =
+    [ctx.local, D.bot ()]
+
+  let combine ctx ?(longjmpthrough = false) (lval:lval option) fexp (f:fundec) (args:exp list) fc (au:D.t) (f_ask:Queries.ask) : D.t =
+    let fromlv = if not longjmpthrough then relevants_from_lval_opt ctx lval else VS.empty () in
+    let taintedcallee = relevants_from_ls (f_ask.f Queries.MayBeTainted) in
```

Comment:
This reminds me that our notion of "tainted" here is not its usual meaning (a la our taint analysis tutorial). We might have to improve this naming sometime separately.